extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,54 +40,173 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
+    this.ampImgEl_ = null;
 
-    /** @private {?string} */
+    /** @private {?number} */
     this.x_ = null;
 
-    /** @private {?string} */
+    /** @private {?number} */
     this.y_ = null;
 
-    /** @private {?string} */
+    /** @private {?number} */
     this.zoom_ = null;
+
+    /** @private {?../../../extensions/amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {?string} */
+    this.pageId_ = null;
+
+    /** @private {boolean} */
+    this.isOnActivePage_ = false;
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.x_ = parseFloat(this.element_.getAttribute('x')) || 0;
+    this.y_ = parseFloat(this.element_.getAttribute('y')) || 0;
+    this.zoom_ = parseFloat(this.element_.getAttribute('zoom')) || 1;
+
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        this.storeService_ = storeService;
+      }
+    );
   }
 
   /** @override */
   layoutCallback() {
-    const ampImgEl = dev().assertElement(
+    this.ampImgEl_ = dev().assertElement(
       this.element_.querySelector('amp-img')
     );
-    return whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    this.initializeListeners_();
+    return whenUpgradedToCustomElement(this.ampImgEl_)
+      .then(() => this.ampImgEl_.signals().whenSignal(CommonSignals.LOAD_END))
       .then(() => {
-        this.image_ = dev().assertElement(this.element.querySelector('img'));
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setImportantStyles(this.ampImgEl_, {
+          left: 'auto',
+          right: 'auto',
+        });
       })
       .catch(() => user().error(TAG, 'Failed to load the amp-img.'));
   }
 
+  /** @private */
+  initializeListeners_() {
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      (currPageId) => {
+        this.isOnActivePage_ = currPageId === this.getPageId_();
+        this.onPageNavigation_();
+      },
+      true /** callToInitialize */
+    );
+    this.storeService_.subscribe(
+      StateProperty.PANNING_MEDIA_STATE,
+      (panningMediaState) => this.onPanningMediaStateChange_(panningMediaState)
+    );
+    // Mutation observer for distance attribute
```

Comment:
Curious why is the MutationObserver necessary? Can't we just get it from the page when the `CURRENT_PAGE_ID` is updated?

Reply:
Reverted, will save for next commit.